src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -198,33 +261,39 @@ static void add_topic_config_if_requested(
           override_name,
           overrides,
           include_synonyms,
+          documentation,
           std::forward<Func>(describe_f));
     }
 }
 
+template<typename T>
+static ss::sstring maybe_print_optional(const std::optional<T>& opt) {
+    return opt ? ssx::sformat("{}", opt.value()) : "-1";
+}
+
 template<typename T>
 static void add_topic_config(
   describe_configs_result& result,
   std::string_view default_name,
   const std::optional<T>& default_value,
   std::string_view override_name,
   const tristate<T>& overrides,
-  bool include_synonyms) {
-    std::optional<ss::sstring> override_value;
-    if (overrides.is_disabled()) {
-        override_value = "-1";
-    } else if (overrides.has_value()) {
-        override_value = ssx::sformat("{}", overrides.value());
+  bool include_synonyms,
+  std::optional<ss::sstring> documentation) {
+    std::optional<std::optional<T>> override_value;
+    if (overrides.has_value()) {
+        override_value = std::make_optional(overrides.value());
```

Comment:
The overrides tristate type is another part of the problem with getting the right typename into `add_topic_config` and thus into `propert_config_type`.

Here my proposal is to use a nested optional but I'm also open to ideas.
I can also talk through my thought process here in zoom or something @graphcareful  @dotnwat but essentially:
- The parent optional represents a "disabled" tristate object
- The child optional represents the actual value

Reply:
I think it would be better to edit the `property_config_type` method and revert these changes here. You can have the `property_config_type` method use recursion to `dive-into` the type `T` in the sole case that it is a tristate. For example:
``` 
if constexpr (is_tristate<T>) {
  return property_config_type<T::inner_type>();
}
```
IMO this is cleaner and will work in all situations where we need to print the config type of any tristate.

Reply:
I don't think I was clear. The only thing that is tristate is `overrides` which we convert to an optional because the original `add_topic_config` defines `overrides` as a `std::optional<T>`. So the issue is about converting the tristate to an optional to satisfy a parameter type.

Reply:
I understand the issue now. Yeah this is tricky. I guess one option would be to remove the `add_topic_config` override (the one that takes a tristate) and push the small amount of logic it had inside of it into the other override. That way you have T, not T as a string.
```
template<typename T, typename Func>
static void add_topic_config(
  describe_configs_result& result,
  std::string_view default_name,
  const T& default_value,
  std::string_view override_name,
  const std::optional<T>& overrides,
  bool include_synonyms,
  std::optional<ss::sstring> documentation,
  Func&& describe_f) {
if constexpr(is_tristate<T>){
    std::optional<ss::sstring> override_value;
    if (overrides.is_disabled()) {
        override_value = "-1";
    } else if (overrides.has_value()) {
        override_value = ssx::sformat("{}", overrides.value());
    }
}
.... etc ...
```

Reply:
I'll give it a try but default value is not a tristate.
```
    add_topic_config(
      result,
      default_name,
      default_value, // This is a std::optional<T>
      override_name,
      override_value, // This is a tristate<T>
      include_synonyms,
      documentation,
      [](const tristate<T>& t) { return overrides.is_disabled() ? "-1" : ssx::sformat("{}", overrides.value()); });
    }
```

Hmm.. Maybe I can instead convert the default_value to tristate?

Reply:
> Hmm.. Maybe I can instead convert the default_value to tristate?

Ignore that, I was thinking out loud

Reply:
Yeah `default_value` is of type `T`.